akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSubscriber.scala

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl.io
+
+import akka.actor.{ ActorLogging, ActorRef, Deploy, Props }
+import akka.stream.actor.{ ActorSubscriberMessage, RequestStrategy }
+import akka.util.ByteString
+
+import scala.concurrent.Promise
+
+/** INTERNAL API */
+object InputStreamSubscriber {
+  def props(completionPromise: Promise[Long], bufSize: Int): Props = {
+    require(bufSize > 0, "buffer size must be > 0")
+    Props(classOf[InputStreamSubscriber], completionPromise, bufSize).withDeploy(Deploy.local)
```

Comment:
He means in the companian object of ```InputStreamSubscriber```,we could make use of the ```
def apply[T <: Actor: ClassTag](creator: ⇒ T): Props``` which is more typesafe and avoid the reflection .Once you changed the actor's constructor,then it will have compile error,and you could detect the compile error quickly:).